### PR TITLE
Chris p social feed

### PR DESCRIPTION
**Pull request by ChrisPichlerEMU**
Defines endpoints, defines logic, and creates database queries for the social feed of the application.

### CODE REVIEW

File: backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/UserProfile.java [hunk 1]
```diff
@@ -33,7 +35,7 @@ public class UserProfile {
 
     //delete (soft delete)
     public boolean isDeleted = false;
-
```
Comment (tvanden3) [inline backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/UserProfile.java:36]: what happened to the spacing here?

File: backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/web/dto/SocialPostDto.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.riptFitness.Ript_Fitness_Backend.domain.model.AccountsModel;
+import com.riptFitness.Ript_Fitness_Backend.domain.model.SocialPostComment;
+
+public class SocialPostDto {
+
+	public Long id;
+	
+	public AccountsModel accountsModel;
+
```
Comment (tvanden3) [inline backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/web/dto/SocialPostDto.java:14]: will you be adding a photo/video attribute to this DTO down the line?

**Review by tvanden3 — COMMENTED** [review]
This is awesome work Chris. You are a gem!

File: backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/web/dto/SocialPostDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.riptFitness.Ript_Fitness_Backend.web.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.riptFitness.Ript_Fitness_Backend.domain.model.AccountsModel;
+import com.riptFitness.Ript_Fitness_Backend.domain.model.SocialPostComment;
+
+public class SocialPostDto {
```
Comment (tvanden3) [inline backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/web/dto/SocialPostDto.java:9]: We could consider changing the fields in the DTO to private for encapsulation? / preventing invalid data

File: backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/AccountsModel.java [hunk 4]
```diff
@@ -34,6 +37,9 @@ protected void onCreate() {
         this.lastLogin = LocalDateTime.now();
     }
```
Comment (tvanden3) [inline backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/AccountsModel.java:39]: potential merge conflict

**Review by tvanden3 — COMMENTED** [review]
Nice work!

File: backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/UserProfile.java [hunk 5]
```diff
@@ -7,7 +7,9 @@
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
 import jakarta.persistence.ElementCollection;
+import jakarta.persistence.CascadeType;
```
Comment (sjiang1) [inline backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/domain/model/UserProfile.java:12]: No other changes in the rest of this file? Why does it need these two new imports?

**Review by esnowgol — APPROVED** [review]
Looks good! Service class code coverage is 94.7% and the Controller class code coverage is 100%

**Comment by sjiang1** [comment]
Quality of Code: Good
Participation in Code Review: Good
Post-Review Changes: Good

(-2 pts): The pull request description needs more details.

**Comment by sjiang1** [comment]
For @ctrotta23 

Quality of Code: Good
Participation in Code Review: Good
Post-Review Changes: Good

(-2 pts): The pull request description needs more details.
(-2 pts): Some commit messages do not follow the required format outlined in the lecture.

**Pull request merged**